### PR TITLE
ref(ts): Convert barChart to typescript

### PR DESCRIPTION
**Pull request by k-fish**
### Summary

Converts barChart to typescript. Modified the incoming type for BarSeries to get around this temporarily as anything using BarSeries should be able to pass in correct (per the docs) options, and we can force the type for the time being

### CODE REVIEW

File: src/sentry/static/sentry/app/components/charts/miniBarChart.tsx [hunk 1]
```diff
@@ -142,7 +151,7 @@ class MiniBarChart extends React.Component<Props> {
           }),
         },
       };
-      series.push(markerSeries);
+      series?.push(markerSeries);
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/charts/miniBarChart.tsx:154]: Is this optional chain needed? `series` should always be an array based on 77 - 101

File: src/sentry/static/sentry/app/views/discover/result/index.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ type ResultProps = {
   savedQuery: SavedQuery | null; // Provided if it's a saved search
   onFetchPage: (nextOrPrev: string) => void;
   onToggleEdit: () => void;
-  utc: boolean | null;
+  utc?: boolean;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/discover/result/index.tsx:46]: When I've touched utc in the past the null|undefined parts were tedious. You might need to keep null in the type to work with the inconsistent typing on this value elsewhere.

File: src/sentry/static/sentry/app/components/charts/miniBarChart.tsx (hunk 1 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/components/charts/miniBarChart.tsx:154]: No, it was a leftover from some code I scrapped. 👍

**Review by markstory — COMMENTED** [review]
Looks good, but there is a conflict.

**Review by markstory — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Pull request merged**